### ISSUE TITLE
[REVIEW] [Accessibility]

### ISSUE THREAD

**Issue by domoritz** [journalovi/2024-luong-graphical-models#15]
### Reviewed version

ec13d0a

### Perceivable

- [ ] Contrast of all text and elements is sufficient. Geometries and large text must have >3:1 contrast against background. Regular text must have >4.5:1.
- [ ] Content has non-visual alternatives. All figures have alt text and videos have transcripts. Equations are screen reader accessible.
- [x] Font sizes for all texts are sufficiently large. Text must not be smaller than 9pt/12px in size. Ideally only minor text is rendered at 9pt (e.g., axis labels) while all other text is larger.
- [x] Where possible within reason, the article works even if I could not perceive colors. Colors should be augmented with patterns, textures, other visual cues, or tooltips if possible.
- [ ] The article can be navigated comfortably with a screen reader. There are no unnecessary accessibility elements (e.g., axis ticks and labels) that clutter the screen reader experience.

### Screen reader

- [x] VoiceOver on macOS
- [ ] VoiceOver on iOS
- [ ] NVDA on Windows
- [ ] JAWS on Windows
- [ ] Not tested with a screen reader (explain below why not)

### Browser

- [x] Safari or other WebKit-based browser
- [ ] Chrome or other Chromium-based browser
- [ ] Firefox

### Comments on Perceivable

Great alt texts overall and great to see that formulas are described nicely as well. 

There seems to be an issue with the `\in` symbol which is very small. 

<img width="386" height="98" alt="Image" src="https://github.com/user-attachments/assets/fe32d176-e845-4abd-aba7-a6953e1bad63" />

Some of the contrasts are very low. For example, the arrows in this image fail contrast checks to the background. 

<img width="780" height="633" alt="Image" src="https://github.com/user-attachments/assets/61c99dc7-d13c-4e87-aed5-ee6682416155" />

The graphs and charts have a lot of elements exposed to the screen reader that are just read out as "image". I'd suggest hiding them with aria-hidden. 

<img width="1080" height="605" alt="Image" src="https://github.com/user-attachments/assets/1d48a257-3a2e-4c53-adb7-3efa032e7a3c" />

<img width="978" height="309" alt="Image" src="https://github.com/user-attachments/assets/8a71990f-3a7f-4819-9921-a776b09d24e8" />

### Operable

- [ ] If there are interactive elements in the article, they can be operated with a keyboard.
- [x] If there are interactive elements in the article, there are textual instructions for how to use them or it's reasonable to assume that it's clear how to use them.
- [x] When navigating over the article with a keyboard, the focus indicator is always visible.
- [x] Interactions that work with a mouse also work on a touch screen (unless explicitly called out).

### Comments on Operable

The nodes in the node-link-diagrams are accessible with a screen reader but not with a keyboard. Would be great to set a tab index so that they can be activated with just a keyboard. 

### Understandable

- [x] All charts have a title and a description.
- [x] Changes in animated or interactive elements are easy to follow.
- [x] All axes have clear labels (either explicit or implicitly given the context).
- [x] There are legends for all charts that need them.

### Comments on Understandable

Great. 

### Robust

- [x] All buttons use the `button` tag.
- [x] Headings, paragraphs, and figures are tagged correctly.

### Comments on Robust

Great

### Assistive

- [x] The data density of all charts is appropriate. Use clustering or other data reduction techniques if elements compete for space but explain the method.
- [x] Navigation and interaction feels intuitive with all interaction methods (mouse, keyboard, touch).
- [x] All visually apparent features and relationships are described.
- [x] Formatting makes values human-readable. Use , or space to separate thousands. Align numbers to the right. Use the smallest, appropriate number of significant digits.

### Comments on Assistive

Great

### Flexible

- [x] The article is readable on a phone.
- [x] The article can be zoomed and font-sizes change appropriately. The layout of the page does not break when zooming in or changing the font size (e.g., using CMD + or CMD - on macOS).
- [x] Long animations (if they exist) can be paused or stopped. Animations should be videos, not GIFs as they cannot be paused.
- [x] Style and charts use a consistent and familiar design (fonts, colors, etc). Interaction defaults should be consistent for all interactive elements.

### Comments on Flexible

Great

### ORCID

0000-0002-3110-1053

**Comment by vuluongj20** [comment]
Hi @domoritz, thank you for the thorough feedback — I really appreciate it! 

Following your recommendations on Perceivable, I’ve:
 -  Fixed the sizing issue with ∈ symbols ([f9d3371](https://github.com/journalovi/2024-luong-graphical-models/commit/f9d3371073457e8f689a04e2514098e7ff6ed23b))
 -  Increased the contrast ratio of lines/arrows ([b09927d](https://github.com/journalovi/2024-luong-graphical-models/commit/b09927dc1fad959ecfae67c6b244920a2f31a9dc)) and heat maps ([ee69525](https://github.com/journalovi/2024-luong-graphical-models/commit/ee695252f8bd7b71cc90530548885142e0b095b9))
 -  Added aria-hidden to child nodes inside SVG graphs and scatter plots ([6e3af14](https://github.com/journalovi/2024-luong-graphical-models/commit/6e3af1428721fa4dc3762eb38c844a355cfc3a28)), leaving only the top-level aria-labels to describe the element

I also realized that there was a bug which left the icons without any descriptive label, and fixed that too ([6bbd3f8](https://github.com/journalovi/2024-luong-graphical-models/commit/6bbd3f8c00d9b16ce7ad1379d888e9c084dcf2ad)).

To your comment on Operable, I just want to double check: [MDN recommends that focusable elements should be interactive.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Guides/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#focusable_elements_should_have_interactive_semantics) The node-link diagrams in this article are not meant to be interactive (* with one exception, see below), rather they are static content similar to images. In that case, should they still be made focusable with tab-index? I’ll defer to you on this.
_\* The only exception is the first node-link diagram, where each node is clickable. Still, I implemented it such that the diagram itself (the `graphics-document` element in the screenshot below) is non-interactive. Keyboard users may interact with the nodes via separate, [visually hidden](https://react-aria.adobe.com/VisuallyHidden) buttons (the `button` elements in the screenshot)._
    <img width="1425" height="297" alt="Image" src="https://github.com/user-attachments/assets/06533f7d-d917-4c30-99b7-672fc67cb942" />

**Comment by domoritz** [comment]
Awesome. Your article was great to begin with and I appreciate the improvements. 

Only interactive nodes should be focusable. I missed the extra buttons. Sounds great as you have it then.

**Comment by vuluongj20** [comment]
@domoritz Got it, I'll keep the node-link diagram as they currently are then! Happy to make more adjustments if you have further recommendations.
